server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -525,4 +531,97 @@ public KnnVectorsFormat getKnnVectorsFormatForField() {
             return new Lucene94HnswVectorsFormat(hnswIndexOptions.m, hnswIndexOptions.efConstruction);
         }
     }
+
+    @Override
+    public SourceLoader.SyntheticFieldLoader syntheticFieldLoader() {
+        if (copyTo.copyToFields().isEmpty() != true) {
+            throw new IllegalArgumentException(
+                "field [" + name() + "] of type [" + typeName() + "] doesn't support synthetic source because it declares copy_to"
+            );
+        }
+        if (indexed) {
+            return new IndexedSyntheticFieldLoader();
+        }
+        return new DocValuesSyntheticFieldLoader();
+    }
+
+    private class IndexedSyntheticFieldLoader implements SourceLoader.SyntheticFieldLoader {
+        private VectorValues values;
+        private boolean hasValue;
+
+        @Override
+        public Stream<Map.Entry<String, StoredFieldLoader>> storedFieldLoaders() {
+            return Stream.of();
+        }
+
+        @Override
+        public DocValuesLoader docValuesLoader(LeafReader leafReader, int[] docIdsInLeaf) throws IOException {
+            values = leafReader.getVectorValues(name());
+            if (values == null) {
+                return null;
+            }
+            return docId -> {
+                hasValue = docId == values.advance(docId);
+                return hasValue;
+            };
+        }
+
+        @Override
+        public boolean hasValue() {
+            return hasValue;
+        }
+
+        @Override
+        public void write(XContentBuilder b) throws IOException {
+            if (false == hasValue) {
+                return;
+            }
+            b.startArray(simpleName());
+            for (float v : values.vectorValue()) {
+                b.value(v);
+            }
+            b.endArray();
+        }
+    }
+
+    private class DocValuesSyntheticFieldLoader implements SourceLoader.SyntheticFieldLoader {
+        private BinaryDocValues values;
+        private boolean hasValue;
+
+        @Override
+        public Stream<Map.Entry<String, StoredFieldLoader>> storedFieldLoaders() {
+            return Stream.of();
+        }
+
+        @Override
+        public DocValuesLoader docValuesLoader(LeafReader leafReader, int[] docIdsInLeaf) throws IOException {
+            values = DocValues.getBinary(leafReader, name());
+            if (values == null) {
```

Comment:
This needs to be `leafReader.getBinary(name())` because `DocValues.getBinary()` never returns `null`